### PR TITLE
Exec the timeout command adaptively according to the alluxio image version

### LINKED ISSUE #765: [BUG] Can't clean cache for Alluxio because of invalid option
**What is your environment(Kubernetes version, Fluid version, etc.)**
- Fluid (edf7060): controller image tag is `v0.6.0-8dcb94d`

**Describe the bug**
Can't clean cache when destroying Alluxio Workers. Log from `alluxioruntime-controller`
```
2021-05-07T15:21:32.906+0800    ERROR   alluxioctl.AlluxioRuntime       operations/base.go:463  Failed  {"alluxioruntime": "default/hbase", "Command": ["timeout", "-t", "60", "alluxio", "fs", "free", "-f", "/"], "FailedReason": "timeout: invalid option -- 't'\nTry 'timeout --help' for more information.", "error": "command terminated with exit code 125"}
github.com/fluid-cloudnative/fluid/vendor/github.com/go-logr/zapr.(*zapLogger).Error
        /go/src/github.com/fluid-cloudnative/fluid/vendor/github.com/go-logr/zapr/zapr.go:128
github.com/fluid-cloudnative/fluid/pkg/ddc/alluxio/operations.AlluxioFileUtils.execWithoutTimeout
        /go/src/github.com/fluid-cloudnative/fluid/pkg/ddc/alluxio/operations/base.go:463
github.com/fluid-cloudnative/fluid/pkg/ddc/alluxio/operations.AlluxioFileUtils.exec.func1
        /go/src/github.com/fluid-cloudnative/fluid/pkg/ddc/alluxio/operations/base.go:444
2021-05-07T15:21:32.906+0800    INFO    alluxioctl.AlluxioRuntime       operations/base.go:450  execute in time {"alluxioruntime": "default/hbase", "command": ["timeout", "-t", "60", "alluxio", "fs", "free", "-f", "/"]}
2021-05-07T15:21:32.906+0800    INFO    alluxioctl.AlluxioRuntime       alluxio/shutdown.go:42  clean cache failed      {"alluxioruntime": "default/hbase", "retry times": 5}
2021-05-07T15:21:32.906+0800    ERROR   alluxioctl.AlluxioRuntime.reconcileRuntimeDeletion      controllers/runtime_controller.go:157   Failed to shutdown the engine   {"alluxioruntime": "default/hbase", "Runtime": {"namespace": "default", "name": "hbase"}, "error": "execute command [timeout -t 60 alluxio fs free -f /] with expectedErr: command terminated with exit code 125 stdout  and stderr timeout: invalid option -- 't'\nTry 'timeout --help' for more information."}
github.com/fluid-cloudnative/fluid/vendor/github.com/go-logr/zapr.(*zapLogger).Error
        /go/src/github.com/fluid-cloudnative/fluid/vendor/github.com/go-logr/zapr/zapr.go:128
github.com/fluid-cloudnative/fluid/pkg/controllers.(*RuntimeReconciler).ReconcileRuntimeDeletion
        /go/src/github.com/fluid-cloudnative/fluid/pkg/controllers/runtime_controller.go:157
github.com/fluid-cloudnative/fluid/pkg/controllers.(*RuntimeReconciler).ReconcileInternal
        /go/src/github.com/fluid-cloudnative/fluid/pkg/controllers/runtime_controller.go:100
github.com/fluid-cloudnative/fluid/pkg/controllers/v1alpha1/alluxio.(*RuntimeReconciler).Reconcile
        /go/src/github.com/fluid-cloudnative/fluid/pkg/controllers/v1alpha1/alluxio/alluxio_runtime_controller.go:97
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:256
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:232
github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
        /go/src/github.com/fluid-cloudnative/fluid/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:211
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1
        /go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait.BackoffUntil
        /go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil
        /go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait.Until
        /go/src/github.com/fluid-cloudnative/fluid/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90
```

It seems `timeout` has no option `-t` which is the cause of this problem.

**What you expect to happen:**
Alluxio destroying works fine with no exceptions.

**How to reproduce it**
1. Create a dataset with AlluxioRuntime as its backend (Any dataset is okay)
2. load some data into Alluxio
3. Delete the dataset and start a cascading deletion for AlluxioRuntime

**Additional Information**

### PR DESCRIPTION
**Pull request by littletiger123**
### Ⅰ. Describe what this PR does
This pr adaptively executes timeout command with proper option according to Alluxio image version.

### Ⅱ. Does this pull request fix one issue?

fixes #765 

### Ⅳ. Describe how to verify it
Test the code in different Alluxio image versions.

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/ddc/alluxio/operations/cached.go [hunk 1]
```diff
@@ -61,17 +61,32 @@ func (a AlluxioFileUtils) CachedState() (cached int64, err error) {
...
+		stdout           string
+		stderr           string
 	)
 
+	stdout, stderr, err = a.exec(releaseVersion, false)
+	if err != nil {
+		err = fmt.Errorf("execute command %v with expectedErr: %v stdout %s and stderr %s", releaseVersion, err, stdout, stderr)
+	}
+
+	if strings.Contains(stdout, "Ubuntu") {
```
Comment (allenhaozi) [inline pkg/ddc/alluxio/operations/cached.go:77]: Will the output be `ubuntu` ?

Comment (cheyang) [reply pkg/ddc/alluxio/operations/cached.go:77]: I also suggest to test, and show sample result in the comments. Thanks very much.

Comment (littletiger123) [reply pkg/ddc/alluxio/operations/cached.go:78]: I have already tested the code yesterday. I checked the code with the alluxio images (release-2.5.0-1,release-2.5.0-2) separately.  The logs of the alluxioruntime controller showed that different commands executed. The logs are like this.

``
 2021-05-22T22:10:28.295+0800    INFO    alluxioctl.AlluxioRuntime       operations/base.go:450  execute in time {"alluxioruntime": "default/hbase", "command": ["timeout", "60", "alluxio", "fs", "free", "-f", "/"]}
``
Thanks for the suggestions.

File: pkg/ddc/alluxio/operations/cached.go [hunk 2]
```diff
@@ -61,17 +61,34 @@ func (a AlluxioFileUtils) CachedState() (cached int64, err error) {
 // clean cache with a preset timeout of 60s
 func (a AlluxioFileUtils) CleanCache(path string) (err error) {
 	var (
-		// todo(TrafalgarZZZ): adaptively exec timeout command with proper option according to Alluxio image version
-		command = []string{"timeout", "60", "alluxio", "fs", "free", "-f", path}
-		stdout  string
-		stderr  string
+		releaseVersion   = []string{"cat", "/etc/issue"}
+		commandForUbuntu = []string{"timeout", "60", "alluxio", "fs", "free", "-f", path}
+		commandForAlpine = []string{"timeout", "-t", "60", "alluxio", "fs", "free", "-f", path}
```
Comment (cheyang) [inline pkg/ddc/alluxio/operations/cached.go:66]: Could you make `commandForUbuntu` and `commandForAlpine` elegant?

**Review by cheyang — CHANGES_REQUESTED** [review]

File: pkg/ddc/alluxio/operations/cached.go [hunk 3]
```diff
@@ -61,17 +61,34 @@ func (a AlluxioFileUtils) CachedState() (cached int64, err error) {
...
+
 	stdout, stderr, err = a.exec(command, false)
 	if err != nil {
 		err = fmt.Errorf("execute command %v with expectedErr: %v stdout %s and stderr %s", command, err, stdout, stderr)
 		return
 	}
 
 	return
-}
+}
```
Comment (cheyang) [inline pkg/ddc/alluxio/operations/cached.go:94]: Add return of line. thanks.

File: pkg/ddc/alluxio/operations/cached.go (hunk 2 above)
Comment (littletiger123) [reply pkg/ddc/alluxio/operations/cached.go]: > Could you make `commandForUbuntu` and `commandForAlpine` elegant?

Thanks. I have fixed it.

File: pkg/ddc/alluxio/operations/cached.go (hunk 3 above)
Comment (littletiger123) [reply pkg/ddc/alluxio/operations/cached.go]: Thanks. All the above are fixed.

**Review by cheyang — APPROVED** [review]

**Pull request merged**